multiapps-controller-process/src/test/java/org/cloudfoundry/multiapps/controller/process/flowable/ProcessActonRegistryTest.java

```diff
@@ -0,0 +1,80 @@
+package org.cloudfoundry.multiapps.controller.process.flowable;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.text.MessageFormat;
+import java.util.List;
+
+import org.cloudfoundry.multiapps.controller.process.Messages;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class ProcessActonRegistryTest {
+
+    private ProcessActionRegistry processActionRegistry;
+
+    @BeforeEach
+    void setUp() {
+        processActionRegistry = new ProcessActionRegistry(getProcessActions());
+    }
+
+    @Test
+    void testGetStartProcessAction() {
+        ProcessAction abortProcessAction = processActionRegistry.getAction(Action.ABORT.getActionId());
+        assertEquals(Action.ABORT, abortProcessAction.getAction());
+    }
+
+    @Test
+    void testGetResumeProcessAction() {
+        ProcessAction resumeProcessAction = processActionRegistry.getAction(Action.RESUME.getActionId());
+        assertEquals(Action.RESUME, resumeProcessAction.getAction());
+    }
+
+    @Test
+    void testGetRetryProcessAction() {
+        ProcessAction retryProcessAction = processActionRegistry.getAction(Action.RETRY.getActionId());
+        assertEquals(Action.RETRY, retryProcessAction.getAction());
+    }
+
+    @Test
+    void testGetAbortProcessAction() {
+        ProcessAction abortProcessAction = processActionRegistry.getAction(Action.ABORT.getActionId());
+        assertEquals(Action.ABORT, abortProcessAction.getAction());
+    }
+
+    @Test
+    void testGetInvalidOperation() {
+        Exception exception = assertThrows(IllegalStateException.class, () -> processActionRegistry.getAction("invalid"));
+        assertEquals(MessageFormat.format(Messages.UNSUPPORTED_ACTION, "invalid"), exception.getMessage());
+    }
+
+    private List<ProcessAction> getProcessActions() {
+        return List.of(new MockedStartProcessAction(), new MockedResumeProcessAction(), new MockedRetryProcessAction(),
```

Comment:
Why do you need these mock classes?
Isn't is possible with `Mockito.spy(StartProcessAction.class)`?

Reply:
Mockito.spy requires 0 arguments constructor